src/traits.rs

```diff
@@ -158,6 +158,19 @@ pub trait Integer:
     /// The value `1`.
     fn one() -> Self;
 
+    /// The value `0` with the same precision as `other`.
+    fn zero_like(other: &Self) -> Self {
+        Self::from_limb_like(Limb::ZERO, other)
+    }
```

Comment:
Perhaps this should be defined on the `Zero` trait?

Reply:
I thought about it, it would have to be defined explicitly for every case then, now it's just a default implementation.

Reply:
It could still have a provided impl with the appropriate bounds, e.g. `AsMut<[Limb]> + Clone`

Reply:
Hm, something like this?
```rust
trait Zero
    ...
    fn zero_like(other: &Self) -> Self where Self: Clone {
        let mut ret = other.clone();
        ret.set_zero();
        ret
    }
}
```

A small downside here is that it will do the unnecessary copy just to fill it with zeros later, but perhaps it's not that impactful in most cases.

Reply:
Yeah, exactly.

It does have that potential disadvantage, however the LLVM optimizer often seems smart enough to elide those writes if it sees they're being subsequently overwritten.

Reply:
Fixed